targets/target_persistent.c

```diff
@@ -36,17 +80,42 @@ int main(int argc, char* argv[]) {
         base_address, code_start, code_end
     };
 
+    // Configure the stalker with a Transformer and EventSink before prefetching
+    // occurs.
+    //
     GumStalkerTransformer* transformer = gum_stalker_transformer_make_from_callback(
         instr_basic_block, &instr_range, NULL);
+    GumEventSink* event_sink = gum_event_sink_make_from_callback(GUM_COMPILE | GUM_BLOCK,
+        prefetch_log, NULL, NULL);
+    gum_stalker_follow_me(stalker, transformer, event_sink);
+    gum_stalker_deactivate(stalker);
 
     static volatile char* branding __attribute__ ((used)) = (char*) "##SIG_AFL_PERSISTENT##";
     char data[1024];
     ssize_t len;
 
     afl_setup();
     afl_start_forkserver();
-    
-    gum_stalker_follow_me(stalker, transformer, NULL);
+
+    // This is written into by "prefetch_log".
+    //
+    prefetch_compiled = g_hash_table_new(NULL, NULL);
```

Comment:
I went with a single hash table as follows:
* The parent creates a hash table of prefetched blocks during init.
* The child inherits a COPY of this on fork.
* The child checks the hash table before writing a block to the IPC
* The child then adds the block to its copy of the hash table to avoid any duplicates being sent.
* The parent reads blocks from the IPC and updates its copy of the hash table as it goes.
* The hash table is then inherited by the next child.